graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/tracing/FederatedTracingInstrumentation.java

```diff
@@ -433,9 +429,30 @@ public Node(Reports.Trace.Node.Builder builder) {
 
     public static class Options {
         private final boolean debuggingEnabled;
+        private final Predicate<ExecutionInput> shouldTracePredicate;
 
-        public Options(boolean debuggingEnabled) {
+        public Options(boolean debuggingEnabled, Predicate<ExecutionInput> shouldTracePredicate) {
             this.debuggingEnabled = debuggingEnabled;
+            this.shouldTracePredicate = shouldTracePredicate;
+        }
+
+        public Options(boolean debuggingEnabled) {
+            this(debuggingEnabled,
+                    // Default implementation:
+                    // If context implements our HTTPRequestHeaders interface, we should only be active
+                    // if the special HTTP header has the special value. If the header isn't provided or has
+                    // a different value, return false - which is interpreted as meaning "don't instrument".
+                    // (If context doesn't implement HTTPRequestHeaders, always instrument.)
+                    (executionInput) -> {
+                        if (executionInput != null && executionInput.getContext() != null) {
+                            Object context = executionInput.getContext();
+                            if (context instanceof HTTPRequestHeaders) {
+                                String header = ((HTTPRequestHeaders) context).getHTTPRequestHeader(FEDERATED_TRACING_HEADER_NAME);
+                                return FEDERATED_TRACING_HEADER_VALUE.equals(header);
+                            }
+                        }
+                        return true;
+                    });
```

Comment:
NOTE: Original implementation as the previous version, only slightly altered due to new way the question is being asked.